backend/private-graph/graph/resolver.go

```diff
@@ -3667,3 +3728,141 @@ func (r *Resolver) GetJiraProjects(
 
 	return jira.GetJiraProjects(workspace, *accessToken)
 }
+
+func (r *Resolver) MicrosoftTeamsBotEndpoint(w http.ResponseWriter, req *http.Request) {
+	fmt.Println("Running MSTEAMS Bot")
+	ctx := req.Context()
+
+	bot, err := microsoft_teams.NewMicrosoftTeamsBot("common")
+
+	if err != nil {
+		log.WithContext(ctx).Error(e.Wrap(err, "error making ms bot adapter"))
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	act, err := bot.ParseRequest(ctx, req)
+
+	if err != nil {
+		log.WithContext(ctx).Error(e.Wrap(err, "error parsing bot framework request"))
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	if act.Type == schema.ConversationUpdate {
+		if len(act.MembersRemoved) > 0 {
+			// identify if our bot is part of the id
+			for _, memberRemoved := range act.MembersRemoved {
```

Comment:
MS teams bot can be installed in multiple "teams/groups" for the same integration (aka tenantID).
So when the bot is removed, we just remove the channels that had previously been saved
belonging to the team/group removed.